### PR TITLE
 HPCC-21879 Enable warnings on eclrtl

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Compiled, Smoketest, OBT

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review. Note, based on another PR that fixes a bug detected during this process.

File: rtl/eclrtl/rtlfield.hpp [hunk 1]
```diff
@@ -220,7 +222,7 @@ struct ECLRTL_API RtlKeyedIntTypeInfo final : public RtlTypeInfoBase
     inline __uint64 getUInt(const void * ptr) const { return (__uint64) getInt(ptr); }
     static unsigned __int64 addBias(__int64 value, unsigned length);
     static __int64 removeBias(unsigned __int64 value, unsigned length);
-    const RtlTypeInfo *child = nullptr;
+    // const RtlTypeInfo *child = nullptr;
```
Comment (ghalliday) [inline rtl/eclrtl/rtlfield.hpp]: Why is this removed?

File: rtl/eclrtl/eclrtl.cpp [hunk 2]
```diff
@@ -558,8 +558,8 @@ double rtlRoundTo(const double x, int places)
 {
     if (x < 0)
         return -rtlRoundTo(-x, places);
-    volatile double tt = x * kk;
-    double x0 = x + tt;
+//    volatile double tt = x * kk;
```
Comment (ghalliday) [inline rtl/eclrtl/eclrtl.cpp]: I suspect the following code should be using x0 instead of x, but not sure if we should change it, or remove these comments?

**Review by ghalliday — COMMENTED** [review]
@richardkchapman a couple of questions.  The 1st commit has been merged.

File: rtl/eclrtl/rtlfield.hpp (hunk 1 above)
Comment (richardkchapman) [reply rtl/eclrtl/rtlfield.hpp]: Because it's unused.

I commented it out rather than delete it as I was unsure whether it SHOULD be used, though I suspect it's not needed.

Comment (ghalliday) [reply rtl/eclrtl/rtlfield.hpp]: I suspect queryChildType() should be implemented for that type.  I would have thought it was useful for walking type information.

**Comment by ghalliday** [comment]
@richardkchapman

**Comment by richardkchapman** [comment]
@ghalliday Moved the keyedInt type info fix to a separate commit (and PR)

Also changed target to 7.4

File: rtl/include/eclhelper_base.hpp [hunk 3]
```diff
@@ -1371,7 +1371,7 @@ class ECLRTL_API CLibraryMemorySpillSplitArg : public CThorSpillArg
 {
 public:
     CLibraryMemorySpillSplitArg(unsigned _tempUsageCount, const char * _filename, IOutputMetaData * _meta) :
-        meta(_meta), filename(_filename), tempUsageCount(_tempUsageCount) {}
+        meta(_meta), filename(_filename), tempUsageCount(_tempUsageCount)  {}
```
Comment (ghalliday) [inline rtl/include/eclhelper_base.hpp]: pedantic: gained a space.

**Comment by ghalliday** [comment]
@richardkchapman please rebase.  One pedantic comment, but happy to merge.

File: rtl/include/eclhelper_base.hpp (hunk 3 above)
Comment (richardkchapman) [reply rtl/include/eclhelper_base.hpp]: I'm stockpiling in case spaces are in short supply after Brexit

**Comment by richardkchapman** [comment]
@ghalliday Rebased and despaced
